config/log_config.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LoggerPatternConfig is an instance of a level specification for a given logger.
+type LoggerPatternConfig struct {
+	Pattern string `json:"pattern"`
+	Level   string `json:"level"`
+}
+
+var loggerPatternRegexp = regexp.MustCompile(`^[a-z_-]+$|^\*$`)
+
+func validatePattern(pattern string) bool {
+	for _, sep := range strings.Split(pattern, ".") {
+		if !loggerPatternRegexp.MatchString(sep) {
+			return false
+		}
+	}
+	return true
+}
+
+// UpdateLoggerRegistry updates the logger registry if necessary  with the specified logConfig.
+func UpdateLoggerRegistry(logConfig []LoggerPatternConfig, loggerRegistry map[string]logging.Logger) (map[string]logging.Logger, error) {
+	newLogRegistry := make(map[string]logging.Logger)
+
+	for _, lpc := range logConfig {
+		if !validatePattern(lpc.Pattern) {
+			return nil, errors.New("Failed to Validate a Pattern")
+		}
+
+		var matcher strings.Builder
+		for idx, ch := range lpc.Pattern {
+			switch ch {
+			case '*':
+				if idx == len(lpc.Pattern)-1 {
+					matcher.WriteString(`\w+(\.\w+)*`)
+				} else {
+					matcher.WriteString(`\w+`)
```

Comment:
`\w+` is not that same character set as what we validate on - we currently allow for literal hyphens `-`.

Reply:
the code should be updated?

Reply:
yeah, it we should be matching on the same inner pattern that we validate on. right now a logger like `a.b-` will pass validation but will not be matched on in this next step